consensus/types/src/application_domain.rs

```diff
@@ -0,0 +1,14 @@
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum ApplicationDomain {
+    Builder,
+}
+
+impl ApplicationDomain {
+    pub fn get_domain_constant(&self) -> u32 {
+        match self {
+            // This value is an application index of 0 with the bitmask applied (so it's equivalent to the bit mask).
+            // Little endian hex: 0x00000001, Binary: 1000000000000000000000000
+            ApplicationDomain::Builder => 16777216,
```

Comment:
It might be nice to have `16777216` as a constant so we don't repeat it in other places of the code. Might even be able to make this function a `const fn`?

Reply:
Went with a constant here for now